packages/flutter/lib/src/widgets/async.dart

```diff
@@ -737,6 +743,17 @@ class FutureBuilder<T> extends StatefulWidget {
   /// [AsyncSnapshot.hasError] will be true.)
   final T? initialData;
 
+  /// The builder used when the [AsyncSnapshot] is in the [ConnectionState.waiting] state.
+  final WidgetBuilder? onPending;
+
+  /// The builder used when the [AsyncSnapshot] has an error.
+  final WidgetBuilder? onError;
```

Comment:
I am not exactly sure what this is enabling people to do that they cannot do today? It seems like it is just making the API more complex?

Reply:
it's less complex if there weren't two APIs, now it's just two API put together which is confusing

Reply:
Potentially. The full rework from https://github.com/flutter/flutter/issues/87399 proposed breaking changes which this PR avoids.